components/discord_webhook/actions/send-message-with-file/send-message-with-file.mjs

```diff
@@ -58,24 +66,19 @@ export default {
       })).data
       : fs.createReadStream(filePath);
 
-    if (includeSentViaPipedream) {
-      if (typeof content !== "string") {
-        content = JSON.stringify(content);
-      }
-      content += `\n\n${this.getSentViaPipedreamText()}`;
+    try {
+      // No interesting data is returned from Discord
+      await this.discordWebhook.sendMessageWithFile({
+        ...params,
```

Comment:
I think we ideally shouldn't pass the prop values (in [`params`](https://github.com/PipedreamHQ/pipedream/blob/0bf71aa27a8649d22b464be6331560fc477b9a75/components/discord_webhook/actions/send-message-with-file/send-message-with-file.mjs#L51-L59)) that aren't used by `sendMessageWithFile` to `sendMessageWithFile`, to make the code more easily understood. (Also applies to the other actions)

Idea/suggestion:
- Include only params used by `sendMessageWithFile` in `params`
- To export the [user input props](https://pipedream.com/docs/components/api/#user-input-props) on failure (as the message), create a method in `send-message-common.mjs` that gets the user input props

For example:
```js
  // `send-message-common.mjs`
  methods: {
    getUserInputProps(omit = [
      "discordWebhook",
    ]) {
      return Object.keys(this)
        .filter((k) => typeof this[k] !== "function" && !omit.includes(k))
        .reduce((props, key) => {
          props[key] = this[key];
          return props;
        }, {});
    },
```
```js
    } catch (err) {
      const unsentMessage = this.getUserInputProps();
      $.export("unsent", unsentMessage);
      throw err;
    }
```

Reply:
Addressed